contracts/SpokePool.sol

```diff
@@ -938,72 +949,78 @@ abstract contract SpokePool is
+
+        // No need to override passed in destination chain id to be equal to chainId(),
+        // since this is not emitted in the FilledRelay event. So we do not have to fear a replay attack.
 
         USSRelayExecution memory relayExecution = USSRelayExecution({
-            relay: USSRelayData({
-                depositor: depositor,
-                recipient: recipient,
-                relayer: exclusiveRelayer,
-                inputToken: inputToken.token,
-                outputToken: outputToken.token,
-                inputAmount: inputToken.amount,
-                outputAmount: outputToken.amount,
-                originChainId: originChainId,
-                destinationChainId: chainId(),
-                depositId: depositId,
-                fillDeadline: fillDeadline,
-                message: message
-            }),
+            relay: relayData,
             relayHash: bytes32(0),
-            updatedOutputAmount: outputToken.amount,
-            updatedRecipient: recipient,
-            updatedMessage: message,
+            updatedOutputAmount: relayData.outputAmount,
+            updatedRecipient: relayData.recipient,
+            updatedMessage: relayData.message,
             repaymentChainId: repaymentChainId,
             slowFill: false,
             payoutAdjustmentPct: 0
         });
         relayExecution.relayHash = keccak256(abi.encode(relayExecution.relay));
 
-        // Validate RelayExecution data
-
-        // Pull output tokens from msg.sender and send to recipient
-
-        // Trigger `message` callback if appropriate.
+        // If a slow fill for this fill was requested then the relayFills value for this hash will be
+        // FillStatus.RequestedSlowFill.
+        bool replacedSlowFillExecution = relayFills[relayExecution.relayHash] == uint256(FillStatus.RequestedSlowFill);
+        _fillRelayUSS(relayExecution);
 
         emit FilledUSSRelay(
-            inputToken,
-            outputToken,
+            relayData.inputToken,
+            relayData.outputToken,
+            relayData.inputAmount,
+            relayData.outputAmount,
             repaymentChainId,
-            originChainId,
-            depositId,
-            fillDeadline,
-            exclusiveRelayer, // or msg.sender
-            depositor,
-            recipient,
-            message,
-            // updatedRecipient,
-            recipient,
-            // slowFill,
-            false,
-            // updatedOutputTokenAmount
-            outputToken.amount,
-            // payout adjustment pct
-            0,
-            // updatedMessage
-            message
+            relayData.originChainId,
+            relayData.depositId,
+            relayData.fillDeadline,
+            relayData.exclusivityDeadline,
+            relayData.exclusiveRelayer,
+            msg.sender,
+            relayData.depositor,
+            relayData.recipient,
+            relayData.message,
+            replacedSlowFillExecution
```

Comment:
I think this flag `replacedSlowFillExecution` could be useful for the dataworker to determine whether to return excess funds to HubPool:
- If `true` then return full `outputAmount` to HubPool, else SlowFill leaf executed and do nothing